.github/workflows/pr-checks.yaml

```diff
@@ -0,0 +1,51 @@
+name: PR checks
+on:
+  push:
+  workflow_dispatch:
+jobs:
+  checks-and-tests:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout the source code
+      uses: actions/checkout@v3
+
+    - name: Install & display rust toolchain
+      run: |
+        rustup show
+        rustup component add rust-src
+
+    - name: Check targets are installed correctly
+      run: rustup target list --installed
+
+    - name: Cache Crates
+      uses: actions/cache@v3
+      with:
+        path: ~/.cargo 
+        key: ${{ runner.os }}-rust-${{ hashFiles('rust-toolchain.toml') }}
+        restore-keys: |
+          ${{ runner.os }}-rust
+
+    - name: Check if cargo-contract exists
+      id: check-cargo-contract
+      continue-on-error: true
+      run: cargo contract --version
+
+    - name: Install cargo contract
+      if: ${{ steps.check-cargo-contract.outcome == 'failure' }}
+      run: |
+        sudo apt-get install binaryen
+        cargo install cargo-dylint dylint-link
+        # Until 2.0.0 official release, we need to specify --version expressly
+        cargo install --force --locked cargo-contract --version 2.0.0-rc
+
+    - name: Compile checks
+      run: |
+        manifest_paths=`find uniswap-v2/contracts -type f -name Cargo.toml`
+        echo $manifest_paths
+        for manifest_path in $manifest_paths; do
+          echo $manifest_path
+          cargo contract check --manifest-path $manifest_path
+        done
+
+    - name: test
+      run: cargo contract test
```

Comment:
From https://github.com/paritytech/cargo-contract/pull/958, `cargo contract test` is removed. Better use `cargo test`

Reply:
@HyunggyuJang This will have to be updated when this repo go to cargo contract 2.0 & ink! 4.0
But right now in cargo contract [v2.0.0-rc.1](https://github.com/paritytech/cargo-contract/releases/tag/v2.0.0-rc.1) this is not removed

Reply:
Got it. Thanks for the clarification!